addons/account/models/account_move.py

```diff
@@ -797,7 +797,7 @@ def _get_pair_to_reconcile(self):
         elif self._context.get('skip_full_reconcile_check') == 'amount_currency_only':
             field = 'amount_residual_currency'
         #target the pair of move in self that are the oldest
-        sorted_moves = sorted(self, key=lambda a: a.date)
+        sorted_moves = sorted(self.with_context(prefetch_fields=False), key=lambda a: a.date)
```

Comment:
did you measure the impact of that single line? 

because, for what I see, you'll end up with 2 queries (one to get the aml dates, one to get aml[field]) instead of currently one (that get all fields of aml but accordingly to rco, there is not that much of an impact since we're not talking about binary or big text fields).